### PR TITLE
Cherry pick ice_in and year_align changes from MOM6-CICE6 repo

### PR DESCRIPTION
**Pull request by anton-seaice**
I cherry picked recent config changes from MOM6-CICE6 repo:

I noted two things when running this:

- Do we need to set the dates in 'ww3_shel.nml' now we have changed them in datm/drof streams xml files?
- The dates on the history output are wrong:
```
ncdump archive/output001/GMOM_JRA_WD.ww3.hi.1900-01-03-00000.nc | grep time
        time = UNLIMITED ; // (1 currently)
        double time(time) ;
                time:units = "seconds since 1900-01-02 00:00:00" ;
                time:calendar = "noleap" ;
        int mapsta(time, ny, nx) ;
        float UAX(time, ny, nx) ;
        float UAY(time, ny, nx) ;
        float ICE(time, ny, nx) ;
        float HS(time, ny, nx) ;
        float T02(time, ny, nx) ;
        float T0M1(time, ny, nx) ;
        float T01(time, ny, nx) ;
        float FP0(time, ny, nx) ;
        float THM(time, ny, nx) ;
        float EF(time, freq, ny, nx) ;
        float USSX(time, ny, nx) ;
        float USSY(time, ny, nx) ;
 time = 172800 ;
```

Note the `time:units` are seconds since 1900-01-02, but the the time is 172800 (= 2 days). This gives a date of 1900-01-04 00:00:00, however the expected result is the output represents the average from 1900-01-02 00:00:00 to 1900-01-02 24:00:00. Is this a known issue @ezhilsabareesh8 ?

(n.b., I set nuopc.runconfig stop and restart to 1 day, and wav history_n to 1 day, and ran twice in Payu)

### CODE REVIEW

**Comment by anton-seaice** [comment]
xarray also interprets this with the 1900-01-04 date

```
array([cftime.DatetimeNoLeap(1900, 1, 4, 0, 0, 0, 0, has_year_zero=True)],
      dtype=object)
```

**Comment by ezhilsabareesh8** [comment]
> * Do we need to set the dates in 'ww3_shel.nml' now we have changed them in datm/drof streams xml files?

For gridded outputs the dates in `ww3_shel.nml` need not to be changed, since the dates from the  `nuopc.runconfig` file will be used for the output.

> * The dates on the history output are wrong:

Upon checking your configuration I also found that the `history_options = nmonths` is not changed to `ndays` when you run the simulation for 1 day. The history options [in these lines](https://github.dev/anton-seaice/MOM6-CICE6-WW3/blob/8d9b6dbe648753c98a3ee3e8a60a7cc558381436/nuopc.runconfig#L434-L436) to be changed.

**Comment by anton-seaice** [comment]
> For gridded outputs the dates in ww3_shel.nml need not to be changed, since the dates from the nuopc.runconfig file will be used for the output.

And non-gridded output? (Can we delete those lines from ww3_shel.nml? 👿) 

> 
> > * The dates on the history output are wrong:
> 
> Upon checking your configuration I also found that the `history_options = nmonths` is not changed to `ndays` when you run the simulation for 1 day. The history options [in these lines](https://github.dev/anton-seaice/MOM6-CICE6-WW3/blob/8d9b6dbe648753c98a3ee3e8a60a7cc558381436/nuopc.runconfig#L434-L436) to be changed.

Yeah I did that, which gave me the daily output above, but that dates in that output are set wrong.

**Comment by ezhilsabareesh8** [comment]
> And non-gridded output? (Can we delete those lines from ww3_shel.nml?)

The point output feature is not fully implemented in the current RYF configuration since the point outputs are binary files and I am currently using it only for testing purposes. The stride is currently set to zero ([refer here](https://github.dev/anton-seaice/MOM6-CICE6-WW3/blob/8d9b6dbe648753c98a3ee3e8a60a7cc558381436/ww3_shel.nml#L25)), preventing the generation of any point outputs. Nevertheless, it's essential to include these lines as we plan to incorporate this feature in the future.

> Yeah I did that, which gave me the daily output above, but the dates in that output are set wrong.

I am trying to reproduce this issue and get back to you shortly.

**Comment by aekiss** [comment]
why is it
```
                time:units = "seconds since 1900-01-02 00:00:00" ;
```
rather than
```
                time:units = "seconds since 1900-01-01 00:00:00" ;
```
?

**Comment by anton-seaice** [comment]
> why is it
> 
> ```
>                 time:units = "seconds since 1900-01-02 00:00:00" ;
> ```
> 
> rather than
> 
> ```
>                 time:units = "seconds since 1900-01-01 00:00:00" ;
> ```
> 
> ?

It looks like the time:units get set based on _current time_ before the model is run:

https://github.com/ESCOMP/WW3/blob/97740a1abb36b7ba6e52a7d6f7a37dd06f915633/model/src/wav_comp_nuopc.F90#L607

but then the _elapsed time_ used for the time value is based on the model start time:

https://github.com/ESCOMP/WW3/blob/97740a1abb36b7ba6e52a7d6f7a37dd06f915633/model/src/wav_comp_nuopc.F90#L1063

Ill investigate more!

**Comment by aekiss** [comment]
So does that mean the units will be different for each successive run of an experiment? I'm wondering whether xarray would handle concatenation in that case.

**Comment by anton-seaice** [comment]
> So does that mean the units will be different for each successive run of an experiment? 

That's what it looks like. e.g., if a run is 1 day long, the first run would be correct (day 1 + 1 day), the second would be day 2 + 2 days (i.e. day 4) , the third would be day 3 + 3 days (i.e. day 6) etc.

> I'm wondering whether xarray would handle concatenation in that case.

I think it handles different time-steps ok, as long as they don't overlap. i.e. I have concatenated monthly and daily data before.

**Comment by anton-seaice** [comment]
I raised https://github.com/ESCOMP/WW3/issues/17

As its specific to the nuopc cap, we might need a patch. I guess we should also raise an issue in the access-om3 repo.

I think I am happy enough this all works otherwise. and the history output problem is unrelated to this change. I suggest we turn on floe size distribution in a subsequent PR (rather than this one).

**Comment by anton-seaice** [comment]
@aekiss @ezhilsabareesh8 :- Can you review please? :)

File: ice_in [hunk 1]
```diff
@@ -1,450 +1,159 @@
...
-  restart_age = .false.
-  restart_fsd = .false.
-  restart_fy = .false.
-  restart_iso = .false.
-  restart_lvl = .false.
-  restart_pond_lvl = .false.
-  restart_pond_topo = .false.
-  restart_snow = .false.
-  tr_aero = .false.
-  tr_fsd = .false.
```
Comment (ezhilsabareesh8) [inline ice_in:75]: `tr_fsd` is required for floe size distribution, we can include it in the next PR

**Review by ezhilsabareesh8 — APPROVED** [review]
It looks good to me.

**Review by aekiss — APPROVED** [review]
Looks good to me, thanks @anton-seaice

**Comment by aekiss** [comment]
Happy for you to merge @anton-seaice

**Comment by anton-seaice** [comment]
I dont have write permission :)

**Pull request merged**